module/fork_spec_test.go

```diff
@@ -475,3 +479,75 @@ func (ts *ForkSpecTestSuite) Test_Success_GetBoundaryEpochs_After_Maxwell_2() {
 	ts.Require().Equal(epochs.PreviousEpochBlockNumber(2000), uint64(1000))
 	ts.Require().Equal(epochs.PreviousEpochBlockNumber(3000), uint64(1000))
 }
+
+func (ts *ForkSpecTestSuite) Test_searchBoundaryHeight_Success() {
+	ctx := context.Background()
+	var currentHeight uint64 = 200
+	var targetTs uint64 = 123_456
+	headerFn := func(ctx context.Context, height uint64) (*types.Header, error) {
+		return &types.Header{Number: big.NewInt(int64(height)), Time: height}, nil
+	}
+	height, err := searchBoundaryHeight(ctx, currentHeight, targetTs, headerFn)
+	ts.NoError(err)
+	ts.Equal(uint64(124), height)
+}
+
+func (ts *ForkSpecTestSuite) Test_searchBoundaryHeight_Error() {
+	ctx := context.Background()
+	var currentHeight uint64 = 200
+	var targetTs uint64 = 123_456
+	headerFn := func(ctx context.Context, height uint64) (*types.Header, error) {
+		return nil, errors.New("test error")
+	}
+	_, err := searchBoundaryHeight(ctx, currentHeight, targetTs, headerFn)
+	ts.Error(err)
+}
+
+func (ts *ForkSpecTestSuite) Test_estimateDistance_NoPrevious() {
+	currentHeader := &types.Header{Number: big.NewInt(int64(123)), Time: 123_456}
+	targetTs := 123_456
+	distance := estimateDistance(nil, currentHeader, uint64(targetTs))
+	ts.Require().Equal(uint64(1), distance)
+}
+
+func (ts *ForkSpecTestSuite) Test_estimateDistance_MoveLowAfterLow() {
+	previousHeader := &types.Header{Number: big.NewInt(int64(200)), Time: 60_000}
+	currentHeader := &types.Header{Number: big.NewInt(int64(100)), Time: 20_000}
+	distance := estimateDistance(previousHeader, currentHeader, 10_987_000)
+	ts.Require().Equal(uint64(23), distance)
+}
+
+func (ts *ForkSpecTestSuite) Test_estimateDistance_MoveHighAfterLow() {
+	previousHeader := &types.Header{Number: big.NewInt(int64(200)), Time: 60_000}
+	currentHeader := &types.Header{Number: big.NewInt(int64(100)), Time: 20_000}
+	distance := estimateDistance(previousHeader, currentHeader, 20_720_000)
+	ts.Require().Equal(uint64(2), distance)
+}
+
+func (ts *ForkSpecTestSuite) Test_estimateDistance_MoveLowAfterHigh() {
```

Comment:
The name might be wrong:

```suggestion
func (ts *ForkSpecTestSuite) Test_estimateDistance_MoveHighAfterHigh() {
```